Port nRF7120 to upstream zephyr

rob-robinson-14 opened the pull request:
Add initial support for nRF7120 SoC: including soc, board and dts support.

rob-robinson-14 committed:
boards: Add Nordic nrf7120dk_nrf7120 board support

Add all board files for nRF7120dk

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

rob-robinson-14 committed:
dts: arm: nordic: Add support for nRF7120

Add dts files for nRF7120 SoC.

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

rob-robinson-14 committed:
soc: nordic: Add initial support for nRF7120 SoC

Add SoC files for nrf7120.

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

nordicjm commented on soc/nordic/nrf71/CMakeLists.txt:
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2025 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library_sources(
```
one line

nordicjm commented on soc/nordic/nrf71/Kconfig:
```diff
@@ -0,0 +1,60 @@
+	default y
+	help
+	  Select this option to power the antenna switch controller when
+	  required, using GPIO P0.09 - Configured in SOC_EARLY_INIT_HOOK.
```
pin configuration is done in devicetree not Kconfig

rob-robinson-14 replied:
Added `zephyr/dts/bindings/arm/nordic,nrf-pwr-antswc.yaml`

cvinayak commented on dts/arm/nordic/nrf7120_enga_cpuapp.dtsi:
```diff
@@ -0,0 +1,148 @@
+
+&bt_hci_sdc {
+	status = "okay";
+};
```
```suggestion
```

rob-robinson-14 commented on boards/nordic/nrf7120dk/nrf7120_cpuapp_common.dtsi:
```diff
@@ -0,0 +1,203 @@
+			reg = <0x10000 DT_SIZE_K(960)>;
+		};
+
+		slot0_ns_partition: partition@100000 {
```
Please verify this has been done as you expect.

dsseng commented on dts/vendor/nordic/nrf7120_enga.dtsi:
```diff
@@ -0,0 +1,957 @@
+	};
+
+	wifi: wifi {
+		compatible = "nordic,nrf7120";
```
Maybe this needs some other compatibile? This one looks like something for the whole SoC and not just its Wi-Fi radio peripheral. Sorry if I'm wrong, I'm not a DT maintainer

```suggestion
		compatible = "nordic,nrf7120-wifi-radio";
```

dsseng commented on soc/nordic/nrf71/Kconfig:
```diff
@@ -0,0 +1,60 @@
+	  This abstraction allows us to configure TRUSTZONE without depending
+	  on peripheral specific symbols.
+
+config NRF_TRUSTZONE_RAM_REGION_SIZE
```
These configs seem to have no references - am I wrong, or they're not used?

rob-robinson-14 commented on soc/nordic/nrf71/Kconfig.defconfig:
```diff
@@ -0,0 +1,45 @@
+
+rsource "Kconfig.defconfig.nrf71*"
+
+if ARM
```
Prefer to maintain consistency with other Nordic platforms.

rob-robinson-14 committed:
modules: tf-m: nordic: Add support for nRF7120 TF-M

This commit adds initial support for non secure nRF7120 targets in
zephyr.

There are important limitations, such as:
- The hardware Crypto accelerator is not supported and thus the non
  secure target is NOT secure for production applications in upstream
  Zephyr.
- The BL2 is not supported, so no DFU is supported with this support

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

rob-robinson-14 committed:
manifest: Update TF-M for nRF7120DK to align with upstream zephyr

Rename nRF7120PDK to nRF7120DK to align with upstream zephyr
Align to nrf4.0
Add tfm_platform_user_memory_ranges.h files

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

rob-robinson-14 committed:
tests: Add support for nRF7120 to CI tests

Initial porting of nRF7120 to zephyr requires support for a number of
tests that are run as part of the CI. This commit adds the support.

Signed-off-by: Robert Robinson <robert.robinson@nordicsemi.no>

nordicjm commented on dts/vendor/nordic/nrf7120_enga.dtsi:
```diff
@@ -0,0 +1,967 @@
+				reg = <0x3e0000 DT_SIZE_K(116)>;
+				erase-block-size = <4096>;
+				write-block-size = <4>;
+			};
```
delete, assign to cpuapp

nordicjm commented on modules/hal_nordic/nrfx/CMakeLists.txt:
```diff
@@ -120,6 +125,7 @@ zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF54HX ${MDK_DIR}/system_nrf54h.
 if(CONFIG_SOC_SERIES_NRF54LX OR CONFIG_SOC_SERIES_BSIM_NRF54LX)
   zephyr_library_sources(${MDK_DIR}/system_nrf54l.c)
 endif()
+zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_NRF71X  ${MDK_DIR}/system_nrf7120_enga.c)
```
double space -> single space

rob-robinson-14 replied:
This is an alignment as with lines 129 and 130.

nordicjm replied:
the alignment of the existing file is wrong

nordicjm commented on boards/nordic/nrf7120dk/board.cmake:
```diff
@@ -0,0 +1,20 @@
+  board_runner_args(jlink "--speed=4000")
+endif()
+
+if(BOARD_NRF7120DK_NRF7120_CPUAPP_NS)
```
so that the boards can be easily duplicated for 3rd party nrf7120 boards without having to change this

tomi-font commented on modules/trusted-firmware-m/nordic/nrf7120_cpuapp/CMakeLists.txt:
```diff
@@ -0,0 +1,28 @@
+target_include_directories(platform_s
+    PUBLIC
+        ${Trusted\ Firmware\ M_SOURCE_DIR}/platform/ext/target/nordic_nrf/nrf7120dk_nrf7120_cpuapp/services/include
+)
```
How is that board-specific? I see the other Nordic boards have this directory as well but they don't have this in their CMakeLists.txt. Is this really necessary? And if so, shouldn't it be done some other way (as the other Nordic boards don't do this)?

rob-robinson-14 replied:
Agreed - `tfm_platform_user_memory_ranges.h` was in dk folder (for all boards) but has now moved to soc folder within TF-M so I think this can be removed.

rob-robinson-14 commented on soc/nordic/soc.yml:
```diff
@@ -45,6 +45,12 @@ family:
               - name: cpurad
               - name: cpuppr
               - name: cpuflpr
+      - name: nrf71
```
Adding the X would deviate from all other nordic devices above. Note 71 series is similar to 53 here in that it is not split in to L/H.

rob-robinson-14 commented on drivers/flash/Kconfig.nrf_mramc:
```diff
@@ -7,6 +7,7 @@
 config SOC_FLASH_NRF_MRAMC
 	bool "Nordic Semiconductor flash driver for MRAM using MRAM Controller"
 	default y
+	depends on !BUILD_WITH_TFM
```
nRF7120 is the first nordic device with MRAMC which is why we haven't seen this so far and so I don't think there is a concern around breakage.


For RRAM the controller and ram are combined together, see: `zephyr/drivers/flash/Kconfig.nrf_rram`
for which: `select NRFX_RRAMC if !BUILD_WITH_TFM` but there is no separate driver for RRAMC.

tomi-font replied:
Oh right, I see.
Actually, now seeing how it's done for RRAM... Sorry for the back and forth but maybe for now let's just stick to how it's done there, so as you had it previously? I don't know if it makes a difference, but probably better to be consistent.

anangl commented on soc/nordic/soc.yml:
```diff
@@ -45,6 +45,12 @@ family:
               - name: cpurad
               - name: cpuppr
               - name: cpuflpr
+      - name: nrf71x
```
I don't think `nrf71x` and `..._NRF71X` is fine. It's nRF71 Series SoC, not nRF71X.

JarmouniA commented on soc/nordic/nrf71/Kconfig.defconfig:
```diff
@@ -0,0 +1,45 @@
+	default !NRF_GRTC_TIMER
+
+config CACHE_NRF_CACHE
+	default y if EXTERNAL_CACHE
```
> This would then deviate from 54l,54h and 92 series?

It that an argument for keeping it!?

JarmouniA replied:
Ah you are selecting EXTERNAL_CACHE by default just below, so no point really in using it as a condition here.

nordicjm commented on boards/nordic/nrf7120dk/nrf7120dk_nrf7120_cpuapp.yaml:
```diff
@@ -0,0 +1,24 @@
+  - zephyr
+sysbuild: true
+ram: 512
+flash: 960
```
flash should be 1992 ? according to slot0_partition size

kartben commented on soc/nordic/nrf71/soc.c:
```diff
@@ -0,0 +1,212 @@
+	*override = (struct mpc_region_override){
+		.config =
+			(nrf_mpc_override_config_t){
+				.slave_number = 0,
```
ditto
